code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/ServiceCreateValidationFilter.java

```diff
@@ -71,6 +76,14 @@ public Object create(String type, ApiRequest request, ResourceManager next) {
         return super.create(type, request, next);
     }
 
+    protected void validateEnvironment(Service service) {
+        Environment env = objectManager.loadResource(Environment.class, service.getEnvironmentId());
+        if (StringUtils.equals(env.getState(), "error")) {
```

Comment:
We should probably read error state from CommonStatesConstants instead of hardcoding it.